registry/storage/validatorstore.go

```diff
@@ -0,0 +1,366 @@
+	if c.operatorID == nil {
+		return nil
+	}
+	committees := c.OperatorCommittees(c.operatorID())
+	var participating []*Committee
+	for _, committee := range committees {
+		if committee.IsParticipating(epoch) {
+			participating = append(participating, committee)
+		}
+	}
+	return participating
+}
+
+func (c *validatorStore) handleSharesAdded(shares ...*types.SSVShare) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	// Update byValidatorIndex
+	for _, share := range shares {
+		if share.HasBeaconMetadata() {
+			c.byValidatorIndex[share.BeaconMetadata.Index] = share
+		}
+
+		// Update byCommitteeID
+		committee := c.byCommitteeID[share.CommitteeID()]
+		if committee == nil {
+			committee = buildCommittee([]*types.SSVShare{share})
+			c.byCommitteeID[committee.ID] = committee
+		} else {
+			committee = buildCommittee(append(committee.Validators, share))
+		}
+		c.byCommitteeID[committee.ID] = committee
+
+		// Update byOperatorID
+		for _, operator := range share.Committee {
+			data := c.byOperatorID[operator.Signer]
+			if data == nil {
+				data = &sharesAndCommittees{
+					shares:     []*types.SSVShare{share},
+					committees: []*Committee{committee},
+				}
+			} else {
+				data.shares = append(data.shares, share)
+				data.committees = append(data.committees, committee)
+			}
+			c.byOperatorID[operator.Signer] = data
+		}
+	}
+}
+
+func (c *validatorStore) handleShareRemoved(pk spectypes.ValidatorPK) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	share := c.byPubKey(pk[:])
+	if share == nil {
+		return
+	}
+
+	// Update byValidatorIndex
+	delete(c.byValidatorIndex, share.BeaconMetadata.Index)
+
+	// Update byCommitteeID
+	committee := c.byCommitteeID[share.CommitteeID()]
+	if committee == nil {
+		return
+	}
+	validators := make([]*types.SSVShare, 0, len(committee.Validators)-1)
+	indices := make([]phase0.ValidatorIndex, 0, len(committee.Validators)-1)
+	for _, validator := range committee.Validators {
+		if validator.ValidatorPubKey != pk {
+			validators = append(validators, validator)
+			indices = append(indices, validator.ValidatorIndex)
+		}
+	}
+	if len(validators) == 0 {
+		delete(c.byCommitteeID, committee.ID)
+	} else {
+		committee.Validators = validators
+		committee.Indices = indices
+	}
+
+	// Update byOperatorID
+	for _, operator := range share.Committee {
+		data := c.byOperatorID[operator.Signer]
+		if data == nil {
+			return
+		}
+		shares := make([]*types.SSVShare, 0, len(data.shares)-1)
+		for _, s := range data.shares {
+			if s.ValidatorPubKey != pk {
+				shares = append(shares, s)
+			}
+		}
+		if len(shares) == 0 {
+			delete(c.byOperatorID, operator.Signer)
+		} else {
+			data.shares = shares
+		}
+	}
+}
+
+func (c *validatorStore) handleShareUpdated(share *types.SSVShare) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	// Update byValidatorIndex
+	if share.HasBeaconMetadata() {
+		c.byValidatorIndex[share.BeaconMetadata.Index] = share
+	}
+
+	// Update byCommitteeID
+	for _, committee := range c.byCommitteeID {
```

Comment:
like you did in shareAdded -
```suggestion
	for _, committee := range c.byCommitteeID {
	if committee.ID != share.CommitteeID() {
	  continue
	}
```

Reply:
https://github.com/ssvlabs/ssv/pull/1495